File: Integrais - unidade 3/SomaRiemann.java
public class SomaRiemann {
    public static void main(String[] args) {
        double n = 100000; // n = número de retangulos
        int a = 2; // a = ponto mais a es
        int b = 5; // b = ponto mais a direita do intervalo
        double soma = 0;
        double deltaX = (b - a) / n;

        for (int i = 0; i < n; i++) {
            double xi = a + i * deltaX;
            soma += f(xi) * deltaX;
        }

        System.out.println("Area total: " + soma);

    }

    public static double f(double x) {
        double funcao = -(x * x) + (4 * x) + 5;
        return funcao;
    }
}

File: Integrais - unidade 3/SomaRiemann.py
import math

def f(x):
  return math.sin(x) + math.exp(1)

def soma_riemann(a, b, n):

  delta_x = (b - a) / n
  soma = 0

  for i in range(n):
    xi = a + i * delta_x
    soma += f(xi) * delta_x

  return soma


a = 0 #intervalo inicial
b = 3 #intervalo final
n = 100000  # numero de retangulo

# Calculate and print the approximate integral value
resultado = soma_riemann(a, b, n)
print("Area total:", resultado)
